source/dialogs/bulkconverterdialog.cpp

```diff
@@ -0,0 +1,221 @@
+#include "bulkconverterdialog.h"
+#include "ui_bulkconverterdialog.h"
+
+#include "formats/powertab/common.h"
+
+#include <app/paths.h>
+#include <app/settingsmanager.h>
+#include <formats/fileformatmanager.h>
+#include <score/score.h>
+
+#include <QFileDialog>
+#include <QFileInfo>
+#include <QDebug>
+#include <QThread>
+
+std::optional<std::string> convertFile(const boost::filesystem::path& src,
+                                       const boost::filesystem::path& dst,
+                                       FileFormatManager& ffm)
+{
+    QFileInfo fileInfo(QString::fromStdString(src.string()));
+    std::optional<FileFormat> format = ffm.findFormat(
+                fileInfo.suffix().toStdString());
+
+    if (!format) return "bad format";
+
+    Score score;
+    try
+    {
+        ffm.importFile(score, src, *format);
+    }
+    catch (const std::exception &e)
+    {
+        return "could not import file: " + src.string();
+    }
+
+    try
+    {
+        ffm.exportFile(score, dst, FileFormat(getPowerTabFileFormat()));
+    }
+    catch (const std::exception &e)
+    {
+        return "could not export file: " + dst.string();
+    }
+
+    return std::nullopt;
+}
+
+BulkConverterWorker::BulkConverterWorker(boost::filesystem::path& source,
+                                         boost::filesystem::path& destination,
+                                         bool dryRun)
+  : QObject(), mySrc(source), myDst(destination), myDryRun(dryRun),
+    myFileCount(0)
+{
+}
+
+BulkConverterWorker::~BulkConverterWorker()
+{
+}
+
+void BulkConverterWorker::walkAndConvert()
+{
+    SettingsManager settings = SettingsManager();
+    FileFormatManager fileFormatManager(settings);
```

Comment:
This could be passed state, unless if it's OK to use these this way.

Reply:
I think the file format manager is okay, but I'm a bit wary of constructing a new settings manager - I don't think there are any guarantees that changes to the settings are immediately visible between different instances, and currently only one is created in powertabeditor.cpp. So, I think you might as well just pass in reference to the main `FileFormatManager` from powertabeditor.cpp

Reply:
Did the change! Thanks for the insight!